packages/sdk/src/lib/dataProtectorCore/getProtectedData.ts

```diff
@@ -109,15 +123,37 @@ export const getProtectedData = async ({
   }
 };
 
-function flattenSchema(schema: DataSchema, parentKey = ''): string[] {
-  return Object.entries(schema).flatMap(([key, value]) => {
-    const newKey = parentKey ? `${parentKey}.${key}` : key;
-    if (typeof value === 'object') {
-      return flattenSchema(value, newKey);
-    } else {
-      return `${newKey}:${value}`;
-    }
-  });
+function flattenSchema(
+  schema: SearchableDataSchema,
+  parentKey?: string
+): { requiredSchemas: string[]; anyOfSchemas: string[][] } {
+  return Object.entries(schema).reduce(
+    (acc, [key, value]) => {
+      const newKey = parentKey ? `${parentKey}.${key}` : key;
+      // any of types
+      if (Array.isArray(value)) {
+        if (value.length > 1) {
+          // one of many
+          acc.anyOfSchemas.push(value.map((entry) => `${newKey}:${entry}`));
+        } else {
+          // one of one
```

Comment:
```suggestion
          // Array of only one type. Similar to single type.
```

Is that correct to say that?